src/main/scala/sigmastate/serialization/ErgoScriptSerializer.scala

```diff
@@ -0,0 +1,28 @@
+package sigmastate.serialization
+
+import sigmastate.SType
+import sigmastate.Values.{Constant, Value}
+import sigmastate.eval.RuntimeIRContext
+
+object ErgoScriptSerializer {
+  def serialize(script: Value[SType]): Array[Byte] = ???
+  def deserialize(bytes: Array[Byte]): Value[SType] = ???
+
+  def extractConstants(script: Value[SType]): (Seq[Constant[_]], Value[SType]) = {
+    val IR: RuntimeIRContext = new RuntimeIRContext {}
```

Comment:
@greenhat context is not lightwight object, so it is better to make is a parameter of the serializer class.